src/plugins/aws/idc/index.ts

```diff
@@ -0,0 +1,199 @@
+  clientCredentials: { clientId: string; clientSecret: string },
+  idc: { id: string; region: string }
+) => {
+  const { clientId, clientSecret } = clientCredentials;
+  const { id, region } = idc;
+
+  return {
+    init: {
+      method: "POST",
+      body: JSON.stringify({
+        clientId,
+        clientSecret,
+        startUrl: `https://${id}.awsapps.com/start`,
+      }),
+    },
+    url: `https://oidc.${region}.amazonaws.com/device_authorization`,
+  };
+};
+
+export const idcLoginSteps: (
+  clientCredentials: { clientId: string; clientSecret: string },
+  idc: { id: string; region: string }
+) => OidcLoginSteps<AWSAuthorizeResponse, AWSTokenResponse> = (
+  clientCredentials: { clientId: string; clientSecret: string },
+  idc: { id: string; region: string }
+) => ({
+  authorize: async () => {
+    const authorizeResponse = await authorize<AWSAuthorizeResponse>(
+      idcAuthorizeRequestBuilder(clientCredentials, idc)
+    );
+    print2(`Please use the opened browser window to continue your idc device authorization.
+  
+      When prompted, confirm that aws idc page displays this code:
+      
+        ${authorizeResponse.userCode}
+      
+      Waiting for authorization...
+      `);
+    void open(authorizeResponse.verificationUriComplete);
+    return authorizeResponse;
+  },
+  activate: async (authorizeResponse) =>
+    await waitForActivation<AWSAuthorizeResponse, AWSTokenResponse>(
+      authorizeResponse
+    )(
+      (authorize) => authorize.expiresIn,
+      (authorize) =>
+        fetchOidcToken<AWSTokenResponse>(
+          idcRequestBuilder(clientCredentials, idc, authorize)
+        )
+    ),
+});
+
+export const assumeRoleWithIdc = async (
+  args: { account?: string; permissionSet: string },
+  idc: { id: string; region: string }
+): Promise<AwsCredentials> =>
+  await cached(
+    `aws-idc-${args.account}-${args.permissionSet}`,
+    async () => {
+      const { region } = idc;
+      // fetch aws client secrets
+      const clientSecrets = await registerClient(region);
+      const oidcResponse = await cached(
+        `aws-idc-device-authorization`,
+        async () => {
+          const data = await oidcLogin(idcLoginSteps(clientSecrets, idc));
+          return { ...data, expiresAt: Date.now() + data.expiresIn * 1e3 };
+        },
+        {
+          duration: 3600e3,
+        },
+        (data) => data.expiresAt < Date.now()
+      );
+      // fetch device authorization
+      const credentials = await exchangeForAWSCredentials(oidcResponse, idc, {
+        account: args.account,
+        permissionSet: args.permissionSet,
+      });
+      return {
+        AWS_ACCESS_KEY_ID: credentials.roleCredentials.accessKeyId,
+        AWS_SECRET_ACCESS_KEY: credentials.roleCredentials.secretAccessKey,
+        AWS_SESSION_TOKEN: credentials.roleCredentials.sessionToken,
+        AWS_SECURITY_TOKEN: credentials.roleCredentials.securityToken ?? "", // portal does not return this value
```

Comment:
I think we set the security token because it's a legacy value that can override `AWS_SESSION_TOKEN`.

It should be set to `AWS_SECURITY_TOKEN: credentials.roleCredentials.sessionToken`. 

See Okta implementation here: https://github.com/p0-security/p0cli/blob/5a4a9d7c731b5590e0756c7e66fc89d619bd6160/src/plugins/aws/assumeRole.ts#L36-L37

We ran into this problem with Venkat before, @gergas3 has more context on this.

Reply:
setting it to session token.